api/rhsm-subscriptions-api-spec.yaml

```diff
@@ -138,8 +142,39 @@ components:
         application/vnd.api+json:
           schema:
             $ref: "#/components/schemas/Errors"
-
   schemas:
+    InsightsResult:
```

Comment:
So I think a couple of small tweaks are necessary here. First, I think that the items in data should be an array of `TallySnapshot`. Also, the other properties currently in `TallyReport` (`product`, `granularity`) should go into `meta`. I think it would be clearer to go ahead and rename `InsightsResult` to `TallyReport`, and then adjust accordingly:

```yaml
    TallyReport:
      properties:
        data:
          type: array
          items:
            $ref: "#/components/schemas/TallySnapshot"
        links:
          type: object
          properties:
            first:
              type: string
            last:
              type: string
            previous:
              type: string
            next:
              type: string
          required:
            - first
            - previous
            - last
            - next
        meta:
          type: object
          properties:
            resultSetSize:
              type: integer
            product:
              type: string
            granularity:
              description: "Describes the significance of each date in the TallySnapshot list. For example if the
                granularity is set to 'weekly', the dates in the TallySnapshot list will represent the start of a
                seven day period."
              type: string
          required:
            - resultSetSize
```
We could go ahead and set up a generic InsightsResult and use the composition pattern outlined in the link if you'd like, I could go either way.

Reply:
Looks like this now:

```json
{
  "data": [
    {
      "date": "2019-07-12T17:48:44.275Z",
      "instance_count": 0,
      "cores": 0
    }
  ],
  "links": {
    "first": "string",
    "last": "string",
    "previous": "string",
    "next": "string"
  },
  "meta": {
    "resultSetSize": 0,
    "product": "string",
    "granularity": "string"
  }
}
```